torch/_dynamo/types.py

```diff
@@ -1,9 +1,59 @@
+import ctypes
 import dataclasses
+import sys
 import types
 from typing import Callable, Dict, List, NamedTuple, Optional, OrderedDict, Union
 
 from typing_extensions import Protocol
 
+if sys.version_info >= (3, 11):
+    # In latest CPython version, the eval frame API was changed to
+    # provide this struct instead of the types.FrameType.
```

Comment:
Is there any stability guarantee for this struct? 🤔  What if we passed in the destructured version from C, at least we would get a build error if the fields change on the struct

Reply:
As is you're just going to corrupt memory if they change the layout here

Reply:
I added a check during the compilation to ensure that the structu from the header still matches this one.